.github/workflows/test.yml

```
@@ -0,0 +1,40 @@
+name: test
+on: [pull_request]
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.13.x, 1.18.x]
+        terraform-version: [0.12.x, 1.1.x]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: ${{ matrix.terraform-version }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Cache modules
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Go Format
+        run: if [ "$(gofmt -l . | grep -v vendor | wc -l)" -gt 0 ]; then exit 1; fi
```

grep -v vendor が必要ならば、前半部分も変更しないほうが良いのではないでしょうか？
変更後のコードだとvendor配下のファイルについてもgofmtが働いてしまう気がします。
```suggestion
        run: if [ "$(gofmt -l `find . -name '*.go' | grep -v vendor` | wc -l)" -gt 0 ]; then exit 1; fi
```

---

失礼致しました。
ご指摘して頂いた通りなので修正致しました。